### PR TITLE
add joinPath method on Matcher

### PR DESCRIPTION
**Pull request by itajaja**

### CODE REVIEW

**Comment by itajaja** [comment]
to be used by other libraries like named-routes

File: src/Matcher.js [hunk 1]
```diff
@@ -39,6 +39,19 @@ export default class Matcher {
     });
   }
 
+  joinPath(basePath, path) {
+    if (!path) {
+      return basePath;
+    }
+    if (path.charAt(0) === '/') {
```
Comment (taion) [inline src/Matcher.js]: this is not supported by this matcher

File: src/Matcher.js [hunk 2]
```diff
@@ -39,6 +39,16 @@ export default class Matcher {
...
   }
 
+  joinPath(basePath, path) {
+    if (!path) {
+      return basePath;
+    }
+    if (basePath.charAt(basePath.length - 1) === '/') {
+      return `${basePath}${path}`;
+    }
+    return `${basePath}/${path}`;
```
Comment (taion) [inline src/Matcher.js]: joining `a`, `/b` will give you `a//b`, which is wrong. you probably want

```js
`${basePath}${this.getCanonicalPath(path)}
```

and strip the trailing slash on `basePath` if one is present

Comment (taion) [inline src/Matcher.js]: might need to canonicalize `basePath` as well in case it's missing the leading slash... maybe not

File: src/Matcher.js [hunk 3]
```diff
@@ -39,6 +39,16 @@ export default class Matcher {
     });
   }
 
+  joinPath(basePath, path) {
```
Comment (taion) [inline src/Matcher.js]: let's call this `joinPaths`

**Pull request merged**